lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -88,84 +89,85 @@ UsdUndoRenameCommand::~UsdUndoRenameCommand()
 {
 }
 
-/*static*/
 UsdUndoRenameCommand::Ptr UsdUndoRenameCommand::create(const UsdSceneItem::Ptr& srcItem, const Ufe::PathComponent& newName)
 {
-	return std::make_shared<UsdUndoRenameCommand>(srcItem, newName);
+    return std::make_shared<UsdUndoRenameCommand>(srcItem, newName);
 }
 
 UsdSceneItem::Ptr UsdUndoRenameCommand::renamedItem() const
 {
-	return _ufeDstItem;
+    return _ufeDstItem;
 }
 
 bool UsdUndoRenameCommand::renameRedo()
 {
-    // Copy the source path using CopySpec, and inactivate the source.
-
-	// We use the source layer as the destination.  An alternate workflow
-	// would be the edit target layer be the destination:
-	// layer = self.fStage.GetEditTarget().GetLayer()
-	bool status = SdfCopySpec(_layer, _usdSrcPath, _layer, _usdDstPath);
-	if (status)
-	{
-        auto srcPrim = _stage->GetPrimAtPath(_usdSrcPath);
-#ifdef UFE_V2_FEATURES_AVAILABLE
-        UFE_ASSERT_MSG(srcPrim, "Invalid prim cannot be inactivated.");
-#else
-        assert(srcPrim);
-#endif
-        status = srcPrim.SetActive(false);
+    // Copy the source path using CopySpec, and remove the source.
+
+    // We use the source layer as the destination.  An alternate workflow
+    // would be the edit target layer be the destination:
+    // _layer = _stage->GetEditTarget().GetLayer()
+    bool status = SdfCopySpec(_layer, _usdSrcPath, _layer, _usdDstPath);
+    if (status) {
+        // remove all scene description for the given path and 
+        // its subtree in the current UsdEditTarget 
+        {
+            UsdEditContext ctx(_stage, _layer);
+            status = _stage->RemovePrim(_ufeSrcItem->prim().GetPath());
+        }
```

Comment:
With the new restriction in place it is now safe to remove the prim from the scene graph rather that deactivating it.When RemovePrim is called, I simply pass the `_ufeSrcItem->prim().GetPath()` rather than creating a new srcPrim which would be really unnecessary.

Redo operation:
1- Call SdfCopySpec to copy ALL the child specs from _usdSrcPath to _usdDstPath. This is a crucial call.
2- Remove the prim at _ufeSrcItem USD path.
3- set _ufeDstItem by calling createSiblingSceneItem
4- send rename notification